visualisations/fan_chart/webviz_fan_chart/__init__.py

```diff
@@ -0,0 +1,134 @@
+        for name, hex in colors.cnames.items():
+            rgbcolors.append(colors.to_rgb(name))
+        print(rgbcolors)
+
+        uniquelines = []
+        lines = []
+
+        for line_id in self.data['name']:
+            if line_id not in uniquelines:
+                uniquelines.append(line_id)
+
+        for line in uniquelines:
+            line_data = self.data[self.data['name'] == line]
+            x = line_data.index.tolist()
+            for column in line_data.columns:
+                if column == 'mean':
+                    lines.append({
+                        'y': line_data[column].tolist(),
+                        'x': x,
+                        'type': 'scatter',
+                        'legendgroup': line,
+                        'name': line,
+                        'mode': 'lines',
+                        'line': {
+                            'color': 'rgba'
+                            + str(rgbcolors[uniquelines.index(line)])[:-1]
+                            + ')'
+                        }
+                    })
+                elif column == 'p90':
+                    lines.append({
+                        'y': line_data[column].tolist()
+                        + line_data['mean'].tolist()[::-1],
+                        'x': x + x[::-1],
+                        'type': 'scatter',
+                        'legendgroup': line,
+                        'name': line,
+                        'fill': 'toself',
+                        'mode': 'lines',
+                        'showlegend': False,
+                        'fillcolor': 'rgba'
+                        + str(rgbcolors[uniquelines.index(line)])[:-1]
+                        + ', 0.6)',
+                        'line': {
+                            'width': 0
+                        }
+                    })
+                elif column == 'p10':
+                    lines.append({
+                        'y': line_data[column].tolist()
+                        + line_data['mean'].tolist()[::-1],
+                        'x': x + x[::-1],
+                        'type': 'scatter',
+                        'legendgroup': line,
+                        'name': line,
+                        'fill': 'toself',
+                        'mode': 'lines',
+                        'showlegend': False,
+                        'fillcolor': 'rgba'
+                        + str(rgbcolors[uniquelines.index(line)])[:-1]
+                        + ', 0.6)',
+                        'line': {
+                            'width': 0
+                        }
+                    })
+                elif column == 'min':
+                    lines.append({
+                        'y': line_data[column].tolist()
+                        + line_data['mean'].tolist()[::-1],
+                        'x': x + x[::-1],
+                        'type': 'scatter',
+                        'legendgroup': line,
+                        'name': line,
```

Comment:
Initialization of shared members should be moved into its own function:

    init_scatter_trace(y, x):
        return {
            'y': y,
            'x': x + x[::-1],
            'type': 'scatter',
            'fill': 'toself',
            ...
        }